components/Toast/react.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import classNames from 'classnames/bind'
+import PropTypes from 'prop-types'
+import { React as Icon } from 'components/Icon'
+import styles from './style.module.scss'
+
+const cx = classNames.bind(styles)
+
+export const Toast = ({ children, headingIcon, heading, buttons, closeToast, toastProps, ...rest }) => (
+  <div {...rest}>
+    <h2>
+      {headingIcon && (
+        <Icon svg={headingIcon} className={cx('toast-header-icon')} size={24} />
+      )}
+      {heading}
+    </h2>
+    <p>{children}</p>
+    {buttons && (
+      <div className={cx('toast-buttons')}>
+        {typeof buttons === 'function' ? buttons(closeToast, toastProps) : buttons}
+      </div>
+    )}
+  </div>
+)
+
+Toast.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
+  headingIcon: PropTypes.string,
+  heading: PropTypes.string,
+  buttons: PropTypes.oneOfType([PropTypes.element, PropTypes.string, PropTypes.func]),
+}
+
+Toast.defaultProps = {
```

Comment:
Since we are moving toward TS and MDL will compile TSX files, this could have used an interface over prop-types.

Reply:
I'm going to hit resolve on this because I didn't intend for it to be a task, but feel free to do this if you want. I don't think this being JS should block it.

Reply:
I went ahead and Typescriptified the whole branch. I didn't realize the main branch was already Typescript-ready - I thought it was just the alpha branch used by RDS.